features/consent/update-addresses-consent-for-account.ts

```diff
@@ -0,0 +1,55 @@
+import { ConverseXmtpClientType } from "@/utils/xmtpRN/client.types";
+import { getXmtpClient } from "@/utils/xmtpRN/sync";
+import logger from "@utils/logger";
+
+type UpdateAddressesConsentForAccountParams = {
+  account: string;
+  addresses: string[];
+  consent: "allow" | "deny";
+};
+
+export const updateAddressesConsentForAccount = async ({
+  account,
+  addresses,
+  consent,
+}: UpdateAddressesConsentForAccountParams): Promise<void> => {
+  const client = (await getXmtpClient(account)) as ConverseXmtpClientType;
+
+  if (!client) {
+    throw new Error("Client not found");
+  }
+
+  logger.debug(
+    `[XMTPRN Contacts] Consenting to addresses on protocol: ${addresses.join(
+      ", "
+    )}`
+  );
+  const start = new Date().getTime();
+
+  if (consent === "allow") {
```

Comment:
nit: just derivce state from consent and perform the loop once with no conditionals